### PR TITLE
Add and modify templates for SNAX

### PR DESCRIPTION
**Pull request by rgantonio**
In this PR, we:

1. Add the SNAX definition to the the schema (located in docs) so that SNAX is disabled by default regardless if it was declared in the core templates or not.
2. Modify the snitch cluster template to add the default SNAX during snitch cluster generation.
3. Add a custom configuration that anyone can use if needed. It also follows the initial architecture SNAX has: 1 SNAX core and 1 DMA core. 

Before building, make sure you create a `generated` directory under the `repo/target/snitch_cluster/` path. Then invoke:
```bash
/repo/target/snitch_cluster/../../util/clustergen.py -c cfg/snax.hjson -o /repo/target/snitch_cluster/generated --wrapper
```
Under the same `repo/starget/snitch_cluster/` path.

@xiaoling-yi Please remind me to teach you why these files are important. Just so that you can get up to speed! I think @JosseVanDelm  knows these already.

NOTE: This PR is next after #13

### CODE REVIEW

File: docs/schema/snitch_cluster.schema.json
Comment (JosseVanDelm) [inline docs/schema/snitch_cluster.schema.json:419]:
```suggestion
                                "xsnax": {
```
Seems useful to see this as a riscv instruction extension

File: docs/schema/snitch_cluster.schema.json
Comment (JosseVanDelm) [inline docs/schema/snitch_cluster.schema.json:422]:
```suggestion
                                    "description": "Enable SNitch Accelerator eXtension (SNAX)",
```
There's a typo in the original

File: target/snitch_cluster/cfg/snax.hjson
Comment (JosseVanDelm) [inline target/snitch_cluster/cfg/snax.hjson:154]:
```suggestion
        xfvec: false,
        xsnax: false,
```

Comment (JosseVanDelm) [reply target/snitch_cluster/cfg/snax.hjson:154]: Seems useful to also disable it here explicitly

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]
Please see my questions.
I think it is mostly about naming, trying to keep it in line with what snitch has used before.

**Comment by JosseVanDelm** [comment]
> Before building, make sure you create a `generated` directory under the `repo/target/snitch_cluster/` path. Then invoke:
> 
> ```shell
> /repo/target/snitch_cluster/../../util/clustergen.py -c cfg/snax.hjson -o /repo/target/snitch_cluster/generated --wrapper
> ```
> 
> Under the same `repo/starget/snitch_cluster/` path.

This seems like a useful time to switch over to automated testing in CI, please add a workflow that runs all examples with the new configuration before merging

File: docs/schema/snitch_cluster.schema.json
Comment (rgantonio) [reply docs/schema/snitch_cluster.schema.json:419]: @JosseVanDelm Is it better to be seen this way? Since we are technically using CRSs instead. I have a feeling that people may get confused if we term it is a RISCV instruction extension where most of what PULP did was to literally add extra instructions.

File: target/snitch_cluster/cfg/snax.hjson
Comment (rgantonio) [reply target/snitch_cluster/cfg/snax.hjson:154]: Then I will also add this on the other templates for clarity.

File: docs/schema/snitch_cluster.schema.json
Comment (JosseVanDelm) [reply docs/schema/snitch_cluster.schema.json:419]: I don't know, the SSRs technically don't add any instructions IMHO?

Comment (rgantonio) [reply docs/schema/snitch_cluster.schema.json:419]: @JosseVanDelm SSRs have their own extensions 🚀 

https://pulp-platform.github.io/snitch_cluster/rm/custom_instructions.html

SSRs are enabled via CSR instruction, BUT configuration is through extensions 🛩️

File: docs/schema/snitch_cluster.schema.json
Comment (rgantonio) [reply docs/schema/snitch_cluster.schema.json]: Fixed!

**Comment by rgantonio** [comment]
@JosseVanDelm and @xiaoling-yi the last 2 commits do the following:

1. First we have a separate test list found under `target/snitch_cluster/sw/test/` called `snax-run.yaml` which is a copy of the `run.yaml` BUT with some tests not included because those tests require the multi-core setting. Or rather, require all 9 cores which we don't do in SNAX.

Here are a few details about the excluded tests. I won't discuss in detail because I can only trace what happened so far. I spent hours trying to debug only to find out these tests require multicore settings. You could see some of the runs in the [github actions log we have](https://github.com/KULeuven-MICAS/snitch_cluster/actions).

- [`event_unit.c`](https://github.com/KULeuven-MICAS/snitch_cluster/blob/ryan/modify-templates/sw/tests/event_unit.c) - this literally waits for an interrupt response from other cores. If you trace the original code, the SNAX (core 0) will be issuing commands to other cores but it expects other compute cores because it skips the DMA core.
- [`openmp_parallel.c`](https://github.com/KULeuven-MICAS/snitch_cluster/blob/ryan/modify-templates/sw/tests/openmp_parallel.c) is waiting for a response from 8 different cores.
- [`openmp_for_static_schedule`](https://github.com/KULeuven-MICAS/snitch_cluster/blob/ryan/modify-templates/sw/tests/openmp_for_static_schedule.c) - this one is harder to trace but the code somewhat implies it's dispatching tasks on 8 threads (8 cores)
- [`team_global.c`](https://github.com/KULeuven-MICAS/snitch_cluster/blob/ryan/modify-templates/sw/tests/team_global.c) - this literally asks if the 9 cores exist so it doesn't make sense at all to run on SNAX.

2. I had to separate the CI for the SNAX test only because it was convenient or easier to test a separate set of tests. Might be useful also for later. I tested if the matrix in CI works, it does but I got stuck figuring out how to separate the tests in a single job run. It's possible if we use conditionals in the .yml, but I figured this is more elegant only because it logs also a separate set of tests for SNAX.

**Comment by rgantonio** [comment]
Bump @JosseVanDelm or @xiaoling-yi please review so we can merge these today and have a playground for tests :)

File: .github/workflows/ci.yml
Comment (JosseVanDelm) [inline .github/workflows/ci.yml:69]:
```suggestion
      - name: Build Hardware
        run: |
          make CFG_OVERRIDE=target/snitch_cluster/cfg/snax.hjson -C target/snitch_cluster bin/snitch_cluster.vlt
      - name: Build Software
        run: |
          make -C target/snitch_cluster sw
```
This is what we use in the Dockerfile in snax-mlir. I know that it essentially does the same as the copy, but it seems just slightly more robust to use the "more abstract API" here and to keep things the same. 
The `lru.hjson` should also used in the later job for the software in this way.

**Review by JosseVanDelm — APPROVED** [review]
Just a minor nit.
@rgantonio this is a great milestone! Congrats!!!!! :rocket: :rocket: :rocket:

File: .github/workflows/ci.yml
Comment (rgantonio) [reply .github/workflows/ci.yml:69]: OK! I shall test this wait.

**Pull request merged**
